addons/web/static/tests/fields/relational_fields_tests.js

```diff
@@ -1237,6 +1237,97 @@ QUnit.module('relational_fields', {
         form.destroy();
     });
 
+    QUnit.test('list in form: name_get with unique ids (default_get)', function (assert) {
+        assert.expect(3);
+
+        var form = createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch: '<form>' +
+                    '<sheet>' +
+                        '<field name="p">' +
+                            '<tree editable="bottom">' +
+                                '<field name="display_name"/>' +
+                                '<field name="trululu"/>' +
+                            '</tree>' +
+                        '</field>' +
+                    '</sheet>' +
+                '</form>',
+            mockRPC: function (route, args) {
+                if (args.method === 'default_get') {
+                    return $.when({
+                        p: [
+                            [0, 0, { display_name: 'record 1', trululu: 1 }],
+                            [0, 0, { display_name: 'record 2', trululu: 1 }]
+                        ]
+                    });
+                }
+                if (args.method === 'name_get') {
+                    assert.deepEqual(args.args[0], _.uniq(args.args[0]),
+                        "should not have duplicates in name_get rpc");
+                }
+                return this._super.apply(this, arguments);
+            },
+        });
+
+        // trululu of 1st record shows 'first record', which is displayed name of trululu with id 1
+        var record1Trululu = form.$('tr.o_data_row').first().find('td.o_data_cell').first().next().text();
+        assert.strictEqual(record1Trululu, "first record",
+            "1st record should have the correct displayed name for trululu field");
+        // trululu of 2nd record shows 'first record', which is displayed name of trululu with id 1
+        var record2Trululu = form.$('tr.o_data_row').first().next().find('td.o_data_cell').first().next().text();
```

Comment:
actually you could check both with one assert like calling text() on a jquery nodeset of several cells (we do that at other places)